File: src/2379. Minimum Recolors to Get K Consecutive Black Blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        char B = 'B';
        int BCount = 0;
        for (int i = 0; i < k; ++i) {
            if (blocks[i] == B) BCount++;
        }
        if (BCount >= k) return 0;
        int maxBCount = BCount;
        for (int i = k; i < blocks.size(); ++i) {
            if (blocks[i] == B) BCount++;
            if (blocks[i - k] == B) BCount--;
            if (BCount >= k) return 0;
            maxBCount = max(maxBCount, BCount);
        }
        return k - maxBCount;
    }
};

File: src/2379. Minimum Recolors to Get K Consecutive Black Blocks.js
/**
 * @param {string} blocks
 * @param {number} k
 * @return {number}
 */

export function minimumRecolors(blocks, k) {
  let BCount = 0
  for (let i = 0; i < k; i++) {
    BCount += blocks[i] === 'B'
  }
  if (BCount >= k) return 0
  let maxBCount = BCount
  for (let i = k; i < blocks.length; i++) {
    BCount += (blocks[i] === 'B') - (blocks[i - k] === 'B')
    if (BCount >= k) return 0
    maxBCount = Math.max(maxBCount, BCount)
  }
  return k - maxBCount
}
